odoo/addons/base/models/ir_default.py

```diff
@@ -188,3 +200,101 @@ def discard_values(self, model_name, field_name, values):
         json_vals = [json.dumps(value, ensure_ascii=False) for value in values]
         domain = [('field_id', '=', field.id), ('json_value', 'in', json_vals)]
         return self.search(domain).unlink()
+
+    def _evaluate_leaf_with_fallback(self, model_name, leaf):
+        """
+        when the field value of the leaf is company_dependent without
+        customization, evaluate if its fallback value will be kept by
+        the leaf
+        return True/False/None(for unknown)
+        """
+        # TODO fix/refactor it when fix/refactor model.filtered_domain
+        (key, comparator, value) = leaf
+        field_name, *rest = key.split('.', 1)
+        model = self.env[model_name]
+        fallback = model._fields[field_name].get_company_dependent_fallback(model)
```

Comment:
What about checking the domain on a new record instead?
```suggestion
        (key, comparator, value) = leaf
        field_name, *rest = key.split('.', 1)
        model = self.env[model_name]
        fallback = model._fields[field_name].get_company_dependent_fallback(model)
        record = model.new()
        record[field_name] = fallback
        return bool(record.filtered_domain([leaf]))
```